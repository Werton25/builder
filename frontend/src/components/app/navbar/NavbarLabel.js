import React from 'react';
import PropTypes from 'prop-types';
import {getCurrentUri} from "../../../selectors/router";
import {connect} from "react-redux";

class NavbarLabel extends React.Component {
  render() {
    return (
      <li>
        <a>{this.props.currentUri}</a>
      </li>
    );
  }
}

NavbarLabel.propTypes = {
  currentUri: PropTypes.string
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state)
});

export default connect(mapStateToProps)(NavbarLabel)
