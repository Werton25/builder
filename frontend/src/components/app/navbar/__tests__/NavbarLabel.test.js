import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NavbarLabel from "../NavbarLabel";

const store = configureStore()({
  router: {location: {pathname: '/service1/method1'}},
  requests: {}
});

describe('NavbarLabel tests', () => {
  it('NavbarLabel renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NavbarLabel/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

