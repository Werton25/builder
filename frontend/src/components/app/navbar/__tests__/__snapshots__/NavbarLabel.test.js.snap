// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLabel tests NavbarLabel renders correctly 1`] = `
<li>
  <a>
    /service1/method1
  </a>
</li>
`;
